db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/scrolling/ScrollingTestBase.java

```diff
@@ -0,0 +1,336 @@
+package com.github.kagkarlsson.scheduler.scrolling;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.github.kagkarlsson.scheduler.DbUtils;
+import com.github.kagkarlsson.scheduler.ExecutionBoundary;
+import com.github.kagkarlsson.scheduler.ScheduledExecution;
+import com.github.kagkarlsson.scheduler.ScheduledExecutionsFilter;
+import com.github.kagkarlsson.scheduler.Scheduler;
+import com.github.kagkarlsson.scheduler.SchedulerClient;
+import com.github.kagkarlsson.scheduler.StopSchedulerExtension;
+import com.github.kagkarlsson.scheduler.TestTasks;
+import com.github.kagkarlsson.scheduler.task.helper.OneTimeTask;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import javax.sql.DataSource;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public abstract class ScrollingTestBase {
```

Comment:
I haven't looked at the specific tests here yet, but am thinking we don't need to run all tests for all supported database engines? Have one detailed functional test (this test-class I assume), which we run against postgres. Then just test some simple `getScheduledExecutions(...)` syntax-test (after, before, limit) against all databases to verify no syntax errors. These are a bit messy, but I think we can squeeze in a testcase [here](https://github.com/kagkarlsson/db-scheduler/blob/master/db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/compatibility/CompatibilityTest.java#L193) for that..

Reply:
I see, thanks for providing this insight. I will then write 2 tests.

Reply:
* Added a compatibility test.
* Reduced the tests in this class to two, one with priority and the other is without. They test the functionality. Only against postgresql.

Please review, if I got it correctly.